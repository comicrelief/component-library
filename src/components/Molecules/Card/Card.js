import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Picture from '../../Atoms/Picture/Picture';

const Container = styled.div`
  display: flex;
  position: relative;
  flex-direction: column;
  background: ${({ theme, backgroundColor }) => theme.color(backgroundColor)};
`;

const Copy = styled.div`
  padding: 20px;
`;

const Card = ({ backgroundColor, image, images, imageAltText, children }) => {
  return (
    <Container backgroundColor={backgroundColor}>
      {image ? (
        <Picture
          alt={imageAltText}
          image={image}
          images={images}
          objectFit="cover"
          width="100%"
          height="100%"
        />
      ) : null}
      <Copy>{children}</Copy>
    </Container>
  );
};

Card.propTypes = {
  backgroundColor: PropTypes.string,
  image: PropTypes.string,
  images: PropTypes.string,
  imageAltText: PropTypes.string,
  children: PropTypes.node.isRequired
};

Card.defaultProps = {
  backgroundColor: 'white',
  image: null,
  images: null,
  imageAltText: ''
};

export default Card;
