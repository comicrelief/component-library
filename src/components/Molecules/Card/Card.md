# Card

```js
const data = require('../../../styleguide/data/data').default;
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';

<Card
  imageLow={data.image}
  images={data.images}
  backgroundColor="yellow"
  height="auto"
>
  <Text tag="h3" color="purple" size="xl">
    Title
  </Text>
  <Text tag="p">Text body copy description</Text>
</Card>;

<Card
  imageLow={data.image}
  images={data.images}
  backgroundColor="yellow"
  height="auto"
  label="Label"
>
  <Text tag="h3" color="purple" size="xl">
    Title
  </Text>
  <Text tag="p">Text body copy description</Text>
</Card>;
```
