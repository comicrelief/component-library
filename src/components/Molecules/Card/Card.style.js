import styled, { css } from 'styled-components';
import spacing from '../../../theme/shared/spacing';

const Container = styled.div`
  position: relative;
  display: flex;

  /* Check for Cards/smallBreakpointRowLayout prop coming from the CMS, if so make horizontal layout */
  @media ${({ theme }) => theme.allBreakpoints('mobile')} {
    flex-direction: ${props => ((props.smallBreakpointRowLayout === true) ? 'row' : 'column')};
  }

  /* Check for Cards/mediumBreakpointRowLayout prop coming from the CMS, if so make horizontal layout */
  @media ${({ theme }) => theme.allBreakpoints('tablet')} {
    flex-direction: ${props => ((props.mediumBreakpointRowLayout === true) ? 'row' : 'column')};
  }

  /* Set desktop and up to normal column view */
  @media ${({ theme }) => theme.allBreakpoints('desktop')} {
    flex-direction: column;
  }

  height: 100%;
  border-radius: ${props => (props.squaredCorners ? '0' : '1rem')};
  overflow: hidden;
  background: ${({ theme, backgroundColor }) => theme.color(backgroundColor)};
`;

const Wrapper = styled.div`
  height: auto;
`;

const Copy = styled.div`
  display: flex;
  flex-direction: column;
  padding: ${spacing('l')};
  /* Check for Cards/smallBreakpointRowLayout prop coming from the CMS, adjust text spacing */
  @media ${({ theme }) => theme.allBreakpoints('mobile')} {
    ${({ smallBreakpointRowLayout }) => (smallBreakpointRowLayout
      === true) && css`
        padding: ${props => ((props.smallBreakpointRowLayout === true) ? `${spacing('sm')}` : `${spacing('l')}`)};
        h1, h2, h3, h4, h5 {
          margin: ${props => ((props.smallBreakpointRowLayout === true) ? `${spacing('sm')} 0 ${spacing('sm')}` : 'inherit')};
        }
      `}
  }
  @media ${({ theme }) => theme.allBreakpoints('tablet')} {
    ${({ mediumBreakpointRowLayout }) => (mediumBreakpointRowLayout
      === true) && css`
        padding: ${props => ((props.smallBreakpointRowLayout === true) ? `${spacing('sm')}` : `${spacing('l')}`)};
        h1, h2, h3, h4, h5 {
          margin: ${props => ((props.smallBreakpointRowLayout === true) ? `${spacing('sm')} 0 ${spacing('sm')}` : 'inherit')};
        }
      `}
  }
  /* Set desktop view to original layout / spacing */
  @media ${({ theme }) => theme.allBreakpoints('desktop')} {
    padding: ${spacing('l')};
    h1, h3, h4, h5 {
      margin: inherit;
    }
    h2 {
      margin: inherit;
    }
    h2:first-child,
    h3:first-child,
    h4:first-child,
    h5:first-child {
      margin: inherit;
    }
  }
`;

export { Container, Wrapper, Copy };
