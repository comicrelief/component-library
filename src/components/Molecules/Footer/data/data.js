export default {
  menuDescription: 'Footer',
  menuGroup: [
    {
      id: 'f7dc5eef-f4eb-5405-8a35-08808b55cb33',
      title: 'Hear from us',
      url: 'https://www.comicrelief.com',
      links: [
        {
          title: 'Get the newsletter',
          url: 'https://www.comicrelief.com/#',
          class: null
        },
        {
          title: 'How we contact you',
          url: 'https://www.comicrelief.com/update-your-preferences',
          class: null
        }
      ]
    },
    {
      id: '9f3980f3-d02b-52e9-ac41-933a778c040a',
      title: 'Get in touch',
      url: 'https://www.comicrelief.com/',
      links: [
        {
          title: 'Contact us',
          url: 'https://www.comicrelief.com/contact-us',
          class: null
        },
        {
          title: 'Your Gift Aid',
          url: 'https://giftaid.comicrelief.com/update',
          class: null
        },
        {
          title: 'FAQs',
          url: 'https://www.comicrelief.com/frequently-asked-questions',
          class: null
        },
        {
          title: 'Red Nose Day 2019 FAQs',
          url:
            'https://www.comicrelief.com/red-nose-days-frequently-asked-questions',
          class: null
        }
      ]
    },
    {
      id: 'eaec5191-bbb3-5e8d-b966-c53fff34998b',
      title: 'About us',
      url: 'https://www.comicrelief.com/',
      links: [
        {
          title: 'Mission',
          url: 'https://www.comicrelief.com/about-comic-relief',
          class: null
        },
        {
          title: 'Our history',
          url: 'https://www.comicrelief.com/about-comic-relief/history',
          class: null
        },
        {
          title: 'Finances',
          url: 'https://www.comicrelief.com/about-comic-relief/finances',
          class: null
        },
        {
          title: 'Partners',
          url: 'https://www.comicrelief.com/working-with-us',
          class: null
        },
        {
          title: 'Meet the team',
          url: 'https://www.comicrelief.com/about-comic-relief/meet-the-team',
          class: null
        },
        {
          title: 'Sport Relief',
          url: 'https://www.sportrelief.com/',
          class: null
        }
      ]
    },
    {
      id: '4a5b055f-1f7e-5f4b-827d-768032fd373c',
      title: 'Careers',
      url: 'https://www.comicrelief.com',
      links: [
        {
          title: 'Working at Comic Relief',
          url: 'https://www.comicrelief.com/careers',
          class: null
        },
        {
          title: 'Open rolls',
          url: 'https://app.beapplied.com/org/comic-relief',
          class: null
        }
      ]
    },
    {
      id: '4a5b055f-1f7e-5f4b-827d-768032fd373d',
      title: 'News',
      url: 'https://www.comicrelief.com',
      links: [
        {
          title: 'News',
          url: 'https://www.comicrelief.com/news',
          class: null
        },
        {
          title: 'Open rolls',
          url: 'https://www.comicrelief.com/press-releases',
          class: null
        }
      ]
    },
    {
      id: 'eaec5191-bbb3-5e8d-b966-c53fff34998a',
      title: 'Legal',
      url: 'https://www.comicrelief.com/',
      links: [
        {
          title: 'Privacy policy',
          url: 'https://www.comicrelief.com/privacy-notice',
          class: null
        },
        {
          title: 'Terms of use',
          url: 'https://www.comicrelief.com/terms-of-use',
          class: null
        },
        {
          title: 'Cookies',
          url: 'https://www.comicrelief.com/cookies-policy',
          class: null
        },
        {
          title: 'Positive practices',
          url: 'https://www.comicrelief.com/positive-practices',
          class: null
        },
        {
          title: 'Text to donate terms of use',
          url: 'https://www.comicrelief.com/legal/sms',
          class: null
        }
      ]
    }
  ]
};
