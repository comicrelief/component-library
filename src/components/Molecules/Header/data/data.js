export default {
  menuDescription: 'I am the navigation',
  menuGroup: [
    {
      id: 'Parent Item Title 1',
      title: 'Parent Item Title 1',
      url: '/parent1',
      links: [
        {
          title: 'Child Item Title 1-1',
          url: '/child-item-1-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 1-2',
          url: '/child-item-1-2',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 1-3',
          url: '/child-item-1-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 2',
      title: 'Parent Item Title 2',
      url: '/parent2',
      links: [
        {
          title: 'Child Item Title 2-1',
          url: '/child-item-2-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 2-2',
          url: '/child-item-2-2',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 2-3',
          url: '/child-item-2-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 3',
      title: 'Parent Item Title 3',
      url: '/parent3',
      links: [
        {
          title: 'Child Item Title 3-1',
          url: '/child-item-3-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 4',
      title: 'Parent Item Title 4',
      url: '/parent4',
      links: [
        {
          title: 'Child Item Title 4-1',
          url: '/child-item-4-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 4-4',
          url: '/child-item-4-4',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 4-4',
          url: '/child-item-4-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 3',
      title: 'Parent Item Title 3',
      url: '/parent3',
      links: [
        {
          title: 'Child Item Title 3-1',
          url: '/child-item-3-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        }
      ]
    }
  ]
};
