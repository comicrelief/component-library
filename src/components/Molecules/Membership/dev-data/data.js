export default {
  cartId: 'test-sept2019-C365',
  regularGiving: {
    otherDescription:
      'Your donation will help us fund amazing projects in the UK and around the world.',
    moneybuys: [
      {
        amount: 10,
        description:
          'a regular supply of toiletries for someone living in a refugee camp in Serbia.'
      },
      {
        amount: 20,
        description:
          'a potentially lifesaving call for a man at risk of suicide in the UK.'
      },
      {
        amount: 30,
        description:
          'the distribution of enough surplus food for a school breakfast club to feed 80 children.'
      }
    ]
  }
};
