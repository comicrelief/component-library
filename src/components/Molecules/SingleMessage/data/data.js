export default {
  title: 'What your money does',
  text:
    'When you donate to Comic Relief or Sport Relief, youâ€™re supporting vulnerable people and communities in the UK and internationally.',
  pictures: {
    small: 'src/components/Atoms/Picture/assets/picture-1.jpg',
    medium: 'src/components/Atoms/Picture/assets/picture-2.jpg',
    large: 'src/components/Atoms/Picture/assets/picture-3.jpg'
  },
  ctaText: 'Find out how'
};
