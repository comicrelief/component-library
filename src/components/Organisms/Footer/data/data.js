export default {
  title: 'Footer',
  menuGroups: [
    {
      title: 'Hear from us',
      id: 'f7dc5eef-f4eb-5405-8a35-08808b55cb33',
      links: [
        {
          title: 'Get the newsletter',
          path: 'https://www.comicrelief.com/#',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'How we contact you',
          path: 'https://www.comicrelief.com/update-your-preferences',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'Get in touch',
      id: '9f3980f3-d02b-52e9-ac41-933a778c040a',
      links: [
        {
          title: 'Contact us',
          path: 'contact-us',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Your Gift Aid',
          path: 'https://giftaid.comicrelief.com/update',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'FAQs',
          path: 'https://www.comicrelief.com/frequently-asked-questions',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Red Nose Day 2019 FAQs',
          path:
            'https://www.comicrelief.com/red-nose-days-frequently-asked-questions',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'About us',
      id: 'eaec5191-bbb3-5e8d-b966-c53fff34998b',
      links: [
        {
          title: 'Mission',
          path: 'https://www.comicrelief.com/about-comic-relief',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Our history',
          path: 'https://www.comicrelief.com/about-comic-relief/history',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Finances',
          path: 'https://www.comicrelief.com/about-comic-relief/finances',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Partners',
          path: 'https://www.comicrelief.com/working-with-us',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Meet the team',
          path: 'https://www.comicrelief.com/about-comic-relief/meet-the-team',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Sport Relief',
          path: 'https://www.sportrelief.com/',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'Careers',
      id: '4a5b055f-1f7e-5f4b-827d-768032fd373c',
      links: [
        {
          title: 'Working at Comic Relief',
          path: 'https://www.comicrelief.com/careers',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Open roles',
          path: 'https://app.beapplied.com/org/comic-relief',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'News',
      id: '4a5b055f-1f7e-5f4b-827d-768032fd373d',
      links: [
        {
          title: 'News',
          path: 'https://www.comicrelief.com/news',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Press area',
          path: 'https://www.comicrelief.com/press-releases',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'Legal',
      id: 'eaec5191-bbb3-5e8d-b966-c53fff34998a',
      links: [
        {
          title: 'Link comp with both URL and Ref',
          url: 'https://www.google.com',
          reference: {
            path: 'homepage'
          },
          internal: {
            type: 'ContentfulComponentLink'
          }
        },
        {
          title: 'Link comp with only Ref',
          url: null,
          reference: {
            path: 'this-is-a-test'
          },
          internal: {
            type: 'ContentfulComponentLink'
          }
        },
        {
          title: 'Link comp with only URL',
          url: 'https://www.yahoo.com',
          reference: null,
          internal: {
            type: 'ContentfulComponentLink'
          }
        },
        {
          title: 'Test whitelisted external link',
          url: 'https://www.sportrelief.com',
          internal: {
            type: 'ContentfulComponentLink'
          }
        },
        {
          title: 'Test non-whitelisted external link',
          url: 'https://bing.com',
          internal: {
            type: 'ContentfulComponentLink'
          }
        },
        {
          title: 'Test non-whitelisted external link',
          url: 'https://bing.com',
          internal: {
            type: 'ContentfulComponentLink'
          }
        }
      ]
    }
  ]
};
