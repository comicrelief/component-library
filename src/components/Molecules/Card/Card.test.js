import React from 'react';
import 'jest-styled-components';
import renderWithTheme from '../../../hoc/shallowWithTheme';
import Card from './Card';
import data from '../../../styleguide/data/data';

it('renders correctly', () => {
  const tree = renderWithTheme(
    <Card image={data.image} images={data.image} backgroundColor="yellow">
      <h2>Title</h2>
      <p>Description</p>
    </Card>
  ).toJSON();

  expect(tree).toMatchInlineSnapshot(`
    .c0 {
      display: -webkit-box;
      display: -webkit-flex;
      display: -ms-flexbox;
      display: flex;
      position: relative;
      -webkit-flex-direction: column;
      -ms-flex-direction: column;
      flex-direction: column;
      height: 100%;
      background: #FFE400;
    }

    .c1 {
      padding: 1rem;
      display: -webkit-box;
      display: -webkit-flex;
      display: -ms-flexbox;
      display: flex;
      -webkit-flex-direction: column;
      -ms-flex-direction: column;
      flex-direction: column;
    }

    <div
      className="c0"
    >
      <div
        className="c1"
      >
        <h2>
          Title
        </h2>
        <p>
          Description
        </p>
      </div>
    </div>
  `);
});
