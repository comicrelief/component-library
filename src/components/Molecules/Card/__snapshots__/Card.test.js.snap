// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  background: #FFE400;
}

.c1 {
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h2>
      Title
    </h2>
    <p>
      Description
    </p>
  </div>
</div>
`;

exports[`renders correctly with no body 1`] = `
.c2 {
  position: relative;
  width: 100%;
  height: 100%;
}

.c3 {
  display: block;
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  background: #FFE400;
}

.c1 {
  height: auto;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      height="100%"
      width="100%"
    >
      <img
        alt=""
        className="c3 lazyload"
        data-lowsrc="http://images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg"
        data-sizes="auto"
        data-src={null}
        data-srcset="http://images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg"
        height="100%"
        src={null}
        srcSet="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        width="100%"
      />
    </div>
  </div>
</div>
`;
